nav2_mppi_controller/include/nav2_mppi_controller/models/trajectories.hpp

```diff
@@ -29,6 +29,7 @@ struct Trajectories
   Eigen::ArrayXXf x;
   Eigen::ArrayXXf y;
   Eigen::ArrayXXf yaws;
+  Eigen::ArrayXf costs;
```

Comment:
You could make this a reference that points to `costs_` (just would need a constructor and pass it in). I didn't trace this completely to know if this is neat in terms of where `Trajectories` is constructed, but it would be preferable so that we didn't have that huge copy

Reply:
I'd also accept another solution that doesn't involve copying the costs vector around. Its quite large.